specs/waku/v2/waku-message.md

```diff
@@ -0,0 +1,86 @@
+---
+title: Waku Message
+version: 2.0.0-alpha1
+status: Raw
+authors: Oskar Thorén <oskar@status.im>
+---
+
+# Table of Contents
+
+- [Abstract](#abstract)
+- [Motivation](#motivation)
+    - [WakuMessage](#wakumessage)
+    - [Protobuf](#protobuf)
+    - [Payload encryption](#payload-encryption)
+        - [Version 0](#version-0)
+        - [Version 1](#version-1)
+- [Differences from Whisper / Waku v1 envelopes](#differences-from-whisper--waku-v1-envelopes)
+- [Changelog](#changelog)
+- [Copyright](#copyright)
+
+# Abstract
+
+This specification provides a way to encapsulate messages sent over Waku with specific information security goals.
+
+# Motivation
+
+When using Waku to send messages over Waku there are multiple concerns:
+- We may have a separate encryption layer as part of our application
+- We may want to provide efficient routing for resource restricted devices
+- We may want to provide compatibility with Waku v1 envelopes
+- We may want payloads to be encrypted by default
+- We may want to provide unlinkability for metadata protection
+
+This specification attempts to provide for these various requirements.
+
+## WakuMessage
+
+A `WakuMessage` is what is being passed around by the other protocols, such as WakuRelay, WakuStore, and WakuFilter.
+
+The `payload` field SHOULD contain whatever payload is being sent. See section below on payload encryption.
+
+The `contentTopic` field SHOULD be filled out to allow for content-based filtering. See [Waku Filter spec](waku-filter.md) for details.
+
+The `version` field SHOULD be filled out to allow for various types of payload encryption.
```

Comment:
I think we can even go with a `MAY` here?

Reply:
True, I guess this would default to 0 regardless